com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/trace/AmbientContext.java

```diff
@@ -0,0 +1,90 @@
+/*******************************************************************************
+* Copyright (c) 2017 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package com.microsoft.java.debug.core.trace;
+
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class AmbientContext extends ConcurrentHashMap<String, Object> {
+    public static final String AMBCTX_ID = AmbientContext.class.getSimpleName() + "Id";
+    static InheritableThreadLocal<AmbientContext> callContext = new InheritableThreadLocal<AmbientContext>() {
```

Comment:
Do we really want to use the name callContext here? This is implying something which behaves just like the .net CallContext.